paramak/shape.py

```diff
@@ -0,0 +1,1210 @@
+        self.azimuth_placement_angle = azimuth_placement_angle
+        self.workplane = workplane
+        self.rotation_axis = rotation_axis
+
+        # neutronics specific properties
+        self.material_tag = material_tag
+        self.tet_mesh = tet_mesh
+
+        self.physical_groups = physical_groups
+
+        # properties calculated internally by the class
+        self.solid = None
+        self.render_mesh = None
+        # self.volume = None
+        self.hash_value = None
+        self.points_hash_value = None
+        self.x_min = None
+        self.x_max = None
+        self.z_min = None
+        self.z_max = None
+        self.graveyard_offset = None  # set by the make_graveyard method
+        self.patch = None
+
+    @property
+    def solid(self):
+        """The CadQuery solid of the 3d object. Returns a CadQuery workplane
+        or CadQuery Compound"""
+
+        ignored_keys = ["_solid", "_hash_value"]
+        if get_hash(self, ignored_keys) != self.hash_value:
+            self.create_solid()
+            self.hash_value = get_hash(self, ignored_keys)
+
+        return self._solid
+
+    @solid.setter
+    def solid(self, value):
+        self._solid = value
+
+    @property
+    def cut(self):
+        return self._cut
+
+    @cut.setter
+    def cut(self, value):
+        self._cut = value
+
+    @property
+    def intersect(self):
+        return self._intersect
+
+    @intersect.setter
+    def intersect(self, value):
+        self._intersect = value
+
+    @property
+    def union(self):
+        return self._union
+
+    @union.setter
+    def union(self, value):
+        self._union = value
+
+    @property
+    def largest_dimension(self):
+        """Calculates a bounding box for the Shape and returns the largest
+        absolute value of the largest dimension of the bounding box"""
+        largest_dimension = 0
+        if isinstance(self.solid, cq.Compound):
+            for solid in self.solid.Solids():
+                largest_dimension = max(
+                    abs(self.solid.BoundingBox().xmax),
+                    abs(self.solid.BoundingBox().xmin),
+                    abs(self.solid.BoundingBox().ymax),
+                    abs(self.solid.BoundingBox().ymin),
+                    abs(self.solid.BoundingBox().zmax),
+                    abs(self.solid.BoundingBox().zmin),
+                    largest_dimension
+                )
+        else:
+            largest_dimension = max(
+                abs(self.solid.val().BoundingBox().xmax),
+                abs(self.solid.val().BoundingBox().xmin),
+                abs(self.solid.val().BoundingBox().ymax),
+                abs(self.solid.val().BoundingBox().ymin),
+                abs(self.solid.val().BoundingBox().zmax),
+                abs(self.solid.val().BoundingBox().zmin),
+                largest_dimension
+            )
+        self.largest_dimension = largest_dimension
+        return largest_dimension
+
+    @largest_dimension.setter
```

Comment:
**SUGGESTION**

The setter sets the `_largest_dimension` attribute but this attribute is never actually used anywhere. Perhaps we always want a just-in-time calculation of the `largest_dimension` property. In that case, there's similarly no need to set the value on Line 192.

<p align="right">
    <a href="https://www.pullrequest.com/">
        <img width="15%" src="https://static.pullrequest.com/images/dashboard/pullrequest-logo.svg">
    </a>
</p>